stream.go

```diff
@@ -0,0 +1,510 @@
+package molecule
+
+import (
+	"bytes"
+	"io"
+	"math"
+
+	"google.golang.org/protobuf/encoding/protowire"
```

Comment:
Is this protowire thing easy to extract? I've been trying to keep this package zero dependency (it has some dependencies but I only use them in tests).

Its probably not a big deal, but its kind of nice that it doesn't depend on what version of the google protobuf library you're using in your own project if you want to use molecule